demisto_sdk/scripts/tests/validate_mypy_global_ignore_test.py

```diff
@@ -0,0 +1,510 @@
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_code)
+
+        git_repo.git_util.stage_file(py_file_path)
+
+        runner = CliRunner()
+
+        with ChangeCWD(git_repo.path):
+            result = runner.invoke(cls.func, [])
+
+        assert result.exit_code == 0
+
+    def test_mypy_ignore_add_not_global(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        a non-global mypy ignore comment is added to the end
+        of a statement.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Type ignore comment is added to the end of a print
+        statement.
+
+        Then:
+        - Exit code 0 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_ignore)
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_mypy_ignore)
+
+        git_repo.git_util.stage_file(py_file_path)
+
+        runner = CliRunner()
+
+        with ChangeCWD(git_repo.path):
+            result = runner.invoke(cls.func, [])
+
+        assert result.exit_code == 0
+
+    def test_mypy_ignore_add_tab(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        a non-global mypy ignore comment is added in the beginning
+        of the statement with a tab.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Type ignore comment is added after a tab.
+
+        Then:
+        - Exit code 0 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_ignore)
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_mypy_ignore_tab)
+
+        git_repo.git_util.stage_file(py_file_path)
+
+        runner = CliRunner()
+
+        with ChangeCWD(git_repo.path):
+            result = runner.invoke(cls.func, [])
+
+        assert result.exit_code == 0
+
+    def test_mypy_ignore_add_global(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        a global mypy ignore comment is added.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Type ignore comment is added globally.
+
+        Then:
+        - Exit code 1 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_global_ignore)
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_mypy_global_ignore)
+
+        git_repo.git_util.stage_file(py_file_path)
```

Comment:
this block of code is repeated in many tests and may be reused/avoided

Reply:
Removed as part of https://github.com/demisto/demisto-sdk/pull/4405/commits/24b66fd5ba7781ebe5267ea8cc5004a29f9b21a1.